core/src/main/kotlin/com/uber/sdk2/core/ui/UberButton.kt

```diff
@@ -0,0 +1,257 @@
+    setPaddingAttributes(context, attrs, defStyleAttr, defStyleRes)
+    setTextAttributes(context, attrs, defStyleAttr, defStyleRes)
+    if (defaultText != 0) {
+      setText(defaultText)
+    }
+  }
+
+  protected val activity: Activity
+    get() {
+      var context = context
+      while (context !is Activity && context is ContextWrapper) {
+        context = context.baseContext
+      }
+
+      if (context is Activity) {
+        return context
+      }
+
+      throw IllegalStateException("Button is not attached to an activity.")
+    }
+
+  @VisibleForTesting
+  @DrawableRes
+  fun getBackgroundResource(): Int {
+    return backgroundResource
+  }
+
+  override fun setBackgroundResource(@DrawableRes backgroundResource: Int) {
+    this.backgroundResource = backgroundResource
+    super.setBackgroundResource(backgroundResource)
+  }
+
+  private fun setBackgroundAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources = intArrayOf(android.R.attr.background)
+    @SuppressLint("ResourceType")
+    val backgroundAttributes =
+      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
+    try {
+      if (backgroundAttributes.hasValue(0)) {
+        val backgroundResource = backgroundAttributes.getResourceId(0, 0)
+        if (backgroundResource != 0) {
+          setBackgroundResource(backgroundResource)
+        } else {
+          setBackgroundColor(backgroundAttributes.getColor(0, Color.BLACK))
+        }
+      } else {
+        setBackgroundColor(backgroundAttributes.getColor(0, Color.BLACK))
+      }
+    } finally {
+      backgroundAttributes.recycle()
+    }
+  }
+
+  @SuppressLint("ResourceType")
+  private fun setDrawableAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources =
+      intArrayOf(
+        android.R.attr.drawableLeft,
+        android.R.attr.drawableTop,
+        android.R.attr.drawableRight,
+        android.R.attr.drawableBottom,
+        android.R.attr.drawablePadding,
+      )
+    val drawableAttributes =
+      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
+    try {
+      setCompoundDrawablesWithIntrinsicBounds(
+        drawableAttributes.getResourceId(0, 0),
+        drawableAttributes.getResourceId(1, 0),
+        drawableAttributes.getResourceId(2, 0),
+        drawableAttributes.getResourceId(3, 0),
+      )
+      compoundDrawablePadding = drawableAttributes.getDimensionPixelSize(4, 0)
+    } finally {
+      drawableAttributes.recycle()
+    }
+  }
+
+  @SuppressLint("ResourceType")
+  private fun setPaddingAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources =
+      intArrayOf(
+        android.R.attr.padding,
```

Comment:
Should we define our own internal padding with pre-defined spacing units?

Reply:
This is carried over from old styles so keeping as-is